src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -891,6 +891,9 @@ internal override TypeSymbol MergeNullability(TypeSymbol other, VarianceKind var
             Debug.Assert(typeA.IsGenericType);
             Debug.Assert(typeA.Equals(typeB, TypeCompareKind.IgnoreDynamicAndTupleNames | TypeCompareKind.IgnoreNullableModifiersForReferenceTypes));
 
+            // Tuple types act as covariant when merging equivalent types.
```

Comment:
>covariant [](start = 34, length = 9)

This is the comment I didn't understand. I believe that tuples behave as whatever variance is requested (we just go ahead and merge the elements with the requested variance).

Reply:
Yes, that is exactly what this means.  When a type parameter is covariant (or it is a tuple type), we merge using the `variance` given to this method, not the variance type of the parameter.  That is the reason for line 912.

---
In reply to: [332161630](https://github.com/dotnet/roslyn/pull/39069#discussion_r332161630) [](ancestors = 332161630)